controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,873 @@
+			loadBalancerSku := containersv211.LoadBalancerSku(spec.LoadBalancerSku)
+			networkProfile.LoadBalancerSku = loadBalancerSku
+		}
+
+		if spec.NetworkPolicy != "" {
+			networkProfile.NetworkPolicy = containersv211.NetworkPolicy(spec.NetworkPolicy)
+		}
+	}
+
+	var agentPoolProfiles []containersv211.ManagedClusterAgentPoolProfile
+	if hasAgentPoolProfile(spec) {
+		for _, np := range spec.NodePools {
+			if np.OrchestratorVersion == "" {
+				np.OrchestratorVersion = spec.KubernetesVersion
+			}
+			agentProfile := containersv211.ManagedClusterAgentPoolProfile{
+				Name:                to.StringPtr(np.Name),
+				Count:               to.Int32Ptr(np.Count),
+				MaxPods:             to.Int32Ptr(np.MaxPods),
+				OsDiskSizeGB:        to.Int32Ptr(np.OsDiskSizeGB),
+				OsType:              containersv211.OSType(np.OsType),
+				VMSize:              containersv211.VMSizeTypes(np.VMSize),
+				Mode:                containersv211.AgentPoolMode(np.Mode),
+				OrchestratorVersion: to.StringPtr(np.OrchestratorVersion),
+			}
+			if len(np.AvailabilityZones) > 0 {
+				agentProfile.AvailabilityZones = to.StringSlicePtr(np.AvailabilityZones)
+			}
+			if hasCustomVirtualNetwork(spec) {
+				agentProfile.VnetSubnetID = vmNetSubnetID
+			}
+			agentPoolProfiles = append(agentPoolProfiles, agentProfile)
+		}
+	} else {
+		agentProfile := containersv211.ManagedClusterAgentPoolProfile{
+			Name:         to.StringPtr("system"),
+			Count:        to.Int32Ptr(1),
+			MaxPods:      to.Int32Ptr(110),
+			OsDiskSizeGB: to.Int32Ptr(128),
+			OsType:       containersv211.Linux,
+			VMSize:       containersv211.StandardDS2V2,
+			Mode:         containersv211.System,
+		}
+		agentPoolProfiles = append(agentPoolProfiles, agentProfile)
+	}
+
+	var linuxProfile *containersv211.LinuxProfile
+	if hasLinuxProfile(spec) {
+		linuxProfile = &containersv211.LinuxProfile{
+			AdminUsername: to.StringPtr(spec.LinuxAdminUsername),
+			SSH: &containersv211.SSHConfiguration{
+				PublicKeys: &[]containersv211.SSHPublicKey{
+					{
+						KeyData: to.StringPtr(spec.LinuxSSHPublicKeyContents),
+					},
+				},
+			},
+		}
+	}
+
+	clientId, clientSecret, err := getSecrets(secretsCache, spec.AzureCredentialSecret)
+	if err != nil {
+		return c, fmt.Errorf("couldn't get secrets with error: %v", err)
+	}
+	managedCluster := containersv211.ManagedCluster{
+		Name:     to.StringPtr(spec.ClusterName),
+		Location: to.StringPtr(spec.ResourceLocation),
+		Tags:     tags,
+		ManagedClusterProperties: &containersv211.ManagedClusterProperties{
+			KubernetesVersion: to.StringPtr(kubernetesVersion),
+			DNSPrefix:         to.StringPtr(dnsPrefix),
+			AgentPoolProfiles: &agentPoolProfiles,
+			LinuxProfile:      linuxProfile,
+			NetworkProfile:    networkProfile,
+			ServicePrincipalProfile: &containersv211.ManagedClusterServicePrincipalProfile{
+				ClientID: to.StringPtr(clientId),
+				Secret:   to.StringPtr(clientSecret),
+			},
+		},
+	}
+
+	if len(spec.PublicAccessSources) > 0 {
```

Comment:
Rename PublicAccessSource to AuthorizedIPRanges if that is what it maps to.

Reply:
+1

Reply:
I have renamed also `PrivateAccess` variable to `PrviateCluster` and `PublicAccessSource` to `AuthorizedIPRanges`